src/box/box.cc

```diff
@@ -3449,15 +3449,20 @@ bootstrap_master(const struct tt_uuid *replicaset_uuid)
  * \pre  master->applier->state == APPLIER_CONNECTED
  * \post master->applier->state == APPLIER_READY
  *
- * @param[out] start_vclock  the vector time of the master
- *                           at the moment of replica bootstrap
+ * \return -1 in case of a transient error
+ *          0 in case everything is fine
  */
-static void
-bootstrap_from_master(struct replica *master)
+static int
+bootstrap_from_master_begin(struct replica *master)
```

Comment:
You always call `bootstrap_from_master_end` after `bootstrap_from_master_begin`. I suggest to keep them one function then.

Reply:
I asked Yan to do that: see [my previous comment](https://github.com/tarantool/tarantool/pull/7183#discussion_r900214694).
The names are quite bad indeed. I don't know. `bootstrap_from_master_retryable` or something similar maybe?

Reply:
It doesn't look good that we are trying to determine the result by whether it throws or not. If it throws, it should throw on any error. And if it returned fine, then it wasn't an error. The second option could be done by an out parameter or something. The function simply doesn't look splittable. These 2 pieces are useless without each other. We can't even find names for them.

Reply:
Ok, I see your point.
Then @yanshtunder, I suggest to make the function look like
```
static bool
bootstrap_from_master(struct replica *master)
```
And when it failed with a transient error, return `false`.
On success return `true`.
And on an unrecoverable error, throw.

Reply:
I did as @sergepetrenko  said